tests/baselines/reference/isolatedModulesAmbientConstEnum.errors.txt

```diff
@@ -1,8 +1,9 @@
-tests/cases/compiler/file1.ts(1,20): error TS1209: Ambient const enums are not allowed when the '--isolatedModules' flag is provided.
+tests/cases/compiler/file1.ts(2,16): error TS2739: Cannot access ambient const enums when the '--isolatedModules' flag is provided.
 
 
 ==== tests/cases/compiler/file1.ts (1 errors) ====
     declare const enum E { X = 1}
-                       ~
-!!! error TS1209: Ambient const enums are not allowed when the '--isolatedModules' flag is provided.
-    export var y;
\ No newline at end of file
+    export var y = E.X;

```

Comment:
Should this error be any other file than the declaration file?

Reply:
The intention with this change is for the error to be on the *access*, not on the *declaration*, so in practice it could be in a completely different file than the const enum declaration. In a typical case, I might import the `chalk` library, which shouldn't error, but any attempt to use the `Level` enum should error.

I could see an argument for putting an additional error on the const enum (like before), but the point of this change is that simply declaring a const enum is *not* an error anymore; the error is if you try to use it.

Reply:
Yes the error is on access but declaring const enum a {} and using a in module1 shouldn't cause error. It should cause when accessing in module2...n instead?

Reply:
Got it. I just tested it a bit more, and I believe the error still makes sense even within the same file. To be clear, this only happens with `isolatedModules` enabled and only happens for *ambient* const enums (the `declare` is necessary in the code above example).

From my testing, it looks like with `isolatedModules` enabled, const enums are *always* compiled as plain enums and const enum accesses are *always* compiled as plain enum accesses, even when the const enum is in the same file. That means that in this code snippet, the first line is erased (since it's a `declare`) and the second line is compiled as-is as `export var y = E.X`. With `isolatedModules`, TypeScript will not inline the constant even within the same file, so this code would crash at runtime (because `E` doesn't exist) even though it would work at runtime without `isolatedModules`. So I think it's fair for TS to error here.

It certainly seems like a reasonable future improvement to implement inlining within the same file even when `isolatedModules` is on. In that case, this error could be limited to cross-file accesses, but for now, even same-file ambient const enum accesses need to be errors. This would also require the Babel implementation to implement same-file constant inlining.